> [AST fuzzer (MSan) — Fuzzer failed (137). See the logs.](https://clickhouse-test-reports.s3.yandex.net/21526/a9313c9eb9acb8920e3a99452ec14c0a116922d5/fuzzer_msan/report.html#fail1)

```
==202==ERROR: MemorySanitizer: allocator is out of memory trying to allocate 0x18 bytes
```

>Integration tests (asan) — fail: 6, passed: 1152, error: 0

`ENOSPC` inner CI issues:

- [test_version_update_after_mutation](https://clickhouse-test-reports.s3.yandex.net/21526/a9313c9eb9acb8920e3a99452ec14c0a116922d5/integration_tests_(asan)/integration_run_test_version_update_after_mutation_test_py_1.txt.out.log)

```
E           Code: 243. DB::Exception: Received from 172.19.0.7:9000. DB::Exception: Cannot reserve 1.00 MiB, not enough space. Stack trace:
```

- [test_adaptive_granularity](https://clickhouse-test-reports.s3.yandex.net/21526/a9313c9eb9acb8920e3a99452ec14c0a116922d5/integration_tests_(asan)/integration_run_test_adaptive_granularity_test_py_1.txt.out.log)

```
E           Code: 243. DB::Exception: Received from 172.18.0.9:9000. DB::Exception: Cannot reserve 1.00 MiB, not enough space. Stack trace:
```

- [test_default_compression_codec](https://clickhouse-test-reports.s3.yandex.net/21526/a9313c9eb9acb8920e3a99452ec14c0a116922d5/integration_tests_(asan)/integration_run_test_default_compression_codec_test_py_1.txt.out.log)

```
2021.03.08 21:16:47.429939 [ 141 ] {ab5e363c-0551-4a10-9d06-813d04d8e231} <Warning> default.compression_table (MergerMutator): Won't merge parts from 3_3_3_0 to 3_3_3_0 because not enough free space: 0.00 B free and unreserved, 14.38 KiB required now (+100% on overhead); suppressing similar warnings for the next hour
```

--- a/src/Storages/StorageDistributed.cpp
+++ b/src/Storages/StorageDistributed.cpp
@@ -530,7 +530,7 @@ void StorageDistributed::read(
         query_info.query, remote_database, remote_table, remote_table_function_ptr);
 
     Block header =
-        InterpreterSelectQuery(query_info.query, context, SelectQueryOptions(processed_stage)).getSampleBlock();
+        InterpreterSelectQuery(query_info.query, context, SelectQueryOptions(processed_stage).analyze()).getSampleBlock();
 
     const Scalars & scalars = context.hasQueryContext() ? context.getQueryContext().getScalars() : Scalars{};
 

--- a/tests/queries/0_stateless/01758_optimize_skip_unused_shards_once.reference
+++ b/tests/queries/0_stateless/01758_optimize_skip_unused_shards_once.reference
@@ -1,0 +1,2 @@
+StorageDistributed (dist_01758): Auto-increment is 0
+StorageDistributed (dist_01758): Skipping irrelevant shards - the query will be sent to the following shards of the cluster (shard numbers): [1]

--- a/tests/queries/0_stateless/01758_optimize_skip_unused_shards_once.sh
+++ b/tests/queries/0_stateless/01758_optimize_skip_unused_shards_once.sh
@@ -1,0 +1,12 @@
+#!/usr/bin/env bash
+
+CLICKHOUSE_CLIENT_SERVER_LOGS_LEVEL=trace
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+$CLICKHOUSE_CLIENT --optimize_skip_unused_shards=1 -nm -q "
+create table dist_01758 as system.one engine=Distributed(test_cluster_two_shards, system, one, dummy);
+select * from dist_01758 where dummy = 0 format Null;
+" |& grep -o "StorageDistributed (dist_01758).*"

